envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/SanUriMatcherFactory.kt

```diff
@@ -0,0 +1,48 @@
+package pl.allegro.tech.servicemesh.envoycontrol.snapshot.resource.listeners.filters
+
+import io.envoyproxy.envoy.type.matcher.RegexMatcher
+import io.envoyproxy.envoy.type.matcher.StringMatcher
+import pl.allegro.tech.servicemesh.envoycontrol.snapshot.TlsAuthenticationProperties
+import java.lang.IllegalArgumentException
+
+class SanUriMatcherFactory(
+    private val tlsProperties: TlsAuthenticationProperties
+) {
+    private val serviceNameTemplate = "{service-name}"
+    private val sanUriWildcardRegex = escapeRegex(tlsProperties.sanUriFormat)
+
+    private fun escapeRegex(sanUriFormat: String): String {
+        val parts = sanUriFormat.split(serviceNameTemplate)
+        if (parts.size != 2) {
+            throw IllegalArgumentException("SAN URI $sanUriFormat does not properly contain $serviceNameTemplate")
+        }
+        val prefix = maybeWrap(parts[0])
+        val suffix = maybeWrap(parts[1])
+        return "$prefix${tlsProperties.serviceNameWildcardRegex}$suffix"
+    }
+
+    private fun maybeWrap(part: String) = if (part != "") """\Q$part\E""" else ""
```

Comment:
Wouldn't it be cleaner if we just use `Regex.quote()`? Anyway I think we don't have to handle empty string differently.

Reply:
It results in a smaller, cleaner regex. I can use `Regex.quote()`

Reply:
As we discussed in private, `Regex.escape()` produces the same input so we will use it. (`Regex.quote()` was a mistake)